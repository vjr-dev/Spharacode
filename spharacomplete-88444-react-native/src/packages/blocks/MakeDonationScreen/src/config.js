// Customizable Area Start
Object.defineProperty(exports, "__esModule", {
    value: true
  });

  exports.DonationGetAPi = "bx_block_donationpayments/donatelists";
  exports.DonationApiContentType = "application/json";
  exports.DonationGetMethodType = "GET";
// Customizable Area End