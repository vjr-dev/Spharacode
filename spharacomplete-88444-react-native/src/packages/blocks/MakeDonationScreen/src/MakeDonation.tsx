// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import React from "react";

import {
    StyleSheet,
    Text,
    View,
    ScrollView,
    Platform,
    
    Image,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    TouchableOpacity,
    TextInput,
    FlatList,
    KeyboardAvoidingView,
    StatusBar
    
} from "react-native";

import MakeDonationController, { Props } from "./MakeDonationController";
import { back1, back2, down_arrow, image_back } from "./assets";
import { RFPercentage, RFValue } from "react-native-responsive-fontsize";
import { heightPercentageToDP as hp } from 'react-native-responsive-screen';
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;
import { COLORS } from "../../../framework/src/Globals";



export default class Donation extends MakeDonationController {
    constructor(props: Props) {
        super(props);
    }

    
    

    render() {
        return (

            <SafeAreaView style={styles.container1}>
                <View style={styles.child}>
                    <ImageBackground
                        source={back1}
                        style={styles.image}>
                        <ImageBackground
                            source={back2}
                            style={styles.image2}>
                            <KeyboardAvoidingView
                                style={{ flex: 1 }}
                                keyboardVerticalOffset={1}
                                behavior={Platform.OS == 'ios' ? 'height' : 'width'}
                            >
                                <StatusBar backgroundColor={"#f17234"} barStyle="default" />
                                <View style={{ height: windowHeight - 220 }}>

                                    <View style={{ height: hp("7%"), width: windowWidth, justifyContent: "space-between", flexDirection: 'row', backgroundColor: COLORS.darkorange, alignItems: "center" }}>
                                        <TouchableOpacity style={{alignItems:"center",height:"90%",justifyContent:"center", width: "15%"}} testID="btnGoBack" onPress={() => this.goback()}>
                                            <Image
                                                source={image_back}
                                                style={{ height: 15, width: 15, tintColor: COLORS.backgroundGray }} />
                                        </TouchableOpacity>
                                        <View style={{ height: "100%", width: "70%", alignItems: "center", justifyContent: "center" }}>
                                            <Text style={{ color: COLORS.backgroundGray,fontWeight: '700',fontSize: RFValue(16)}}>Make Donation</Text>
                                        </View>
                                        <View style={{ height: "100%", width: "15%"}} />
                                    </View>
                                    <ScrollView>
                                    <View
                                        style={styles.view1}
                                    >
                                        <Text style={styles.whitecolor}>Donate to</Text>

                                    </View>

                                    {this.state.DModal ? <TouchableOpacity
                                        testID="modaltoggleID"
                                        onPress={() => this.setState({
                                            DModal: false
                                        })}
                                        style={styles.donatbtn}
                                    >
                                        <Text style={styles.donattext}>{this.state.Donatt_to}</Text>
                                        <Image
                                            source={down_arrow}
                                            style={styles.arowimage}
                                        />

                                    </TouchableOpacity> : <View
                                        style={styles.view2}
                                    >
                                        <View style={styles.view3}>
                                            <Text style={styles.text1}>Donate to</Text>
                                        </View>
                                        <View style={styles.line} />
                                        <FlatList
                                            data={this.state.final_list}
                                            // style={{ backgroundColor: "pink" }}
                                            renderItem={({ item }: any) => {
                                                return (
                                                    <TouchableOpacity
                                                        onPress={() => this.flatclick(item)}
                                                        style={{ marginLeft: RFPercentage(2.5), marginTop: RFValue(3) }}>
                                                        <Text style={{ color: COLORS.lightwhite }}>{item}</Text>
                                                    </TouchableOpacity>
                                                )
                                            }}
                                        />
                                    </View>}

                                    <View
                                        style={styles.view4}
                                    >
                                        <Text style={styles.whitecolor}>Amount</Text>

                                    </View>
                                    <View
                                        style={styles.view5}
                                    >
                                        <TouchableOpacity
                                            testID="staticamount1ID"
                                            onPress={() => this.staticamount1()}
                                            style={{ height: "100%", width: "23%", backgroundColor: this.state.SAmountstatus == 1 ? COLORS.orangelight : COLORS.Viewback, alignItems: "center", justifyContent: "center", borderRadius: 25 }}
                                        >
                                            <Text style={{ color: this.state.SAmountstatus == 1 ? COLORS.black : COLORS.ultralightwhite }}>₹ 100</Text>
                                        </TouchableOpacity>
                                        <TouchableOpacity
                                        testID="staticamount2ID"
                                            onPress={() => this.staticamount2()}
                                            style={{ height: "100%", width: "23%", backgroundColor: this.state.SAmountstatus == 2 ? COLORS.orangelight : COLORS.Viewback, alignItems: "center", justifyContent: "center", borderRadius: 25 }}
                                        >
                                            <Text style={{ color: this.state.SAmountstatus == 2 ? COLORS.black : COLORS.ultralightwhite }}>₹ 500</Text>
                                        </TouchableOpacity>
                                        <TouchableOpacity
                                            testID="staticamount3ID"
                                            onPress={() => this.staticamount3()}
                                            style={{ height: "100%", width: "23%", backgroundColor: this.state.SAmountstatus == 3 ? COLORS.orangelight : COLORS.Viewback, alignItems: "center", justifyContent: "center", borderRadius: 25 }}
                                        >
                                            <Text style={{ color: this.state.SAmountstatus == 3 ? COLORS.black : COLORS.ultralightwhite }}>₹ 1000</Text>
                                        </TouchableOpacity>
                                        <TouchableOpacity
                                            testID="staticamount4ID"
                                            onPress={() => this.staticamount4()}
                                            style={{ height: "100%", width: "23%", backgroundColor: this.state.SAmountstatus == 4 ? COLORS.orangelight : COLORS.Viewback, alignItems: "center", justifyContent: "center", borderRadius: 25 }}
                                        >
                                            <Text style={{ color: this.state.SAmountstatus == 4 ? COLORS.black : COLORS.ultralightwhite }}>₹ 2000</Text>
                                        </TouchableOpacity>
                                    </View>
                                    <View
                                        style={styles.view6}
                                    >
                                        <TextInput
                                            value={this.state.Amount+""}
                                            testID="customAmountID"
                                            maxLength={10}
                                            keyboardType="number-pad"
                                            style={{ marginLeft: 20, color: COLORS.ultralightwhite }}
                                            placeholderTextColor={COLORS.ultralightwhite}
                                            placeholder="Enter custom amount"
                                            onChangeText={(TT) => this.coustomamount(TT)}

                                        />
                                    </View>
                                    </ScrollView>
                                </View>
                                <TouchableOpacity
                                testID="doneclickID"
                                    onPress={() => this.donatclick()}
                                    style={styles.view7}
                                >
                                    <Text style={{ color: COLORS.white,fontWeight:"700" }}>DONATE NOW</Text>
                                </TouchableOpacity>


                            </KeyboardAvoidingView>
                        </ImageBackground>
                    </ImageBackground>
                </View>
            </SafeAreaView>

            
        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: windowHeight, width: windowWidth,
        backgroundColor: COLORS.white
    },
    container1: { flex: 1, backgroundColor: "#f17234" },
    child: { width: windowWidth, },

    image: {
        height: windowHeight, width: windowWidth,
    },
    image2: {
        height: '100%', width: windowWidth, alignItems: "center",
    },
    view1: { height: 30, width: windowWidth - 20, alignSelf: "center", marginTop: 40, justifyContent: "center" },
    whitecolor: { color: COLORS.lightwhite },
    donatbtn: { height: 50, width: windowWidth - 20, alignSelf: "center", backgroundColor: COLORS.Viewback, borderRadius: 50, flexDirection: "row", justifyContent: "space-between", alignItems: "center" },
    donattext: { color: COLORS.ultralightwhite, marginLeft: 15 },
    arowimage: { height: 15, width: 15, tintColor: COLORS.ultralightwhite, marginRight: 18 },
    view2: { height: 200, width: windowWidth - 20, alignSelf: "center", backgroundColor: COLORS.Viewback, borderRadius: 20 },
    view3: { height: 50, width: "100%", justifyContent: "center" },
    text1: { color: COLORS.ultralightwhite, marginLeft: 20 },
    line: { height: 1, width: "90%", backgroundColor: COLORS.ultralightwhite, alignSelf: "center" },
    view4: { height: 30, width: windowWidth - 20, alignSelf: "center", marginTop: 50, justifyContent: "center" },
    view5: {
        height: 50,
        width: windowWidth - 20,
        alignSelf: "center",
        marginTop: 20,
        justifyContent: "space-between",
        flexDirection: "row"
    },
    view6: { height: 50, width: windowWidth - 20, alignSelf: "center", backgroundColor: COLORS.Viewback, borderRadius: 50, justifyContent: "center", marginTop: 50 },
    view7: { height: 50, width: windowWidth - 20, alignSelf: "center", backgroundColor: "rgba(255,138,73,1)", borderRadius: 50, marginBottom: 20, alignItems: "center", justifyContent: "center" },
});
// Customizable Area End
