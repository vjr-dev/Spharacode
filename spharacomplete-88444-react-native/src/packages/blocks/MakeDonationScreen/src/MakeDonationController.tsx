// Customizable Area Start
//@ts-nocheck
//@ts-ignore
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
    getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

import AsyncStorage from '@react-native-async-storage/async-storage'
import { Alert } from "react-native";



export const configJSON = require("./config");


export interface Props {
    navigation: any;
    id: string;
    
    
}

interface S {
    
    data: any;
    DModal: boolean;
    SAmountstatus: number;
    Amount: any;
    token: any;
    Donation_list: any;
    final_list: any;
    Donatt_to: any

    
}

interface SS {
    id: any;
    
    
}

export default class MakeDonationController extends BlockComponent<Props, S, SS> {
    
    DonationGetId: any;
    

    constructor(props: Props) {
        super(props);
        this.receive = this.receive.bind(this);

        
        this.subScribedMessages = [
            
            getName(MessageEnum.RestAPIResponceMessage),
            getName(MessageEnum.RestAPIResponceDataMessage),
            getName(MessageEnum.RestAPIResponceSuccessMessage),
            getName(MessageEnum.RestAPIResponceErrorMessage),
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            getName(MessageEnum.RestAPIRequestMessage),
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            getName(MessageEnum.RestAPIRequestMethodMessage),
            
        ];

        this.state = {
            
            data: [1, 2, 3, 4, 5],
            DModal: true,
            SAmountstatus: 0,
            Amount: 0,
            token: "",
            Donation_list: "",
            final_list: [],
            Donatt_to: "you are more concerned to"
            
        };
        runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);

        
        
    }

    
    //istanbul ignore next
    async componentDidMount() {

        let TOKEN = await AsyncStorage.getItem("Token")
        this.setState({ token: TOKEN })

        this.GetDonationlist()



    }
    flatclick(item: any) {
        this.setState({ DModal: true, Donatt_to: item })

    }
    donatclick() {
        if (this.state.Amount != 0) {
            this.props.navigation.navigate("PaymentDonation", { Amount: this.state.Amount, Dto: this.state.Donatt_to })
        } else {
            Alert.alert("Please Enter Amount")
        }
    }

    coustomamount(TT: any) {
        this.setState({ Amount: TT, SAmountstatus: 0 })
        console.log(this.state.Amount, TT);

    }

    staticamount1() {
        this.setState({ SAmountstatus: 1, Amount: 100 })
        console.log("1", this.state.Amount);
    }
    staticamount2() {
        this.setState({ SAmountstatus: 2, Amount: 500 })
        console.log("2", this.state.Amount);

    }
    staticamount3() {
        this.setState({ SAmountstatus: 3, Amount: 1000 })
        console.log("3", this.state.Amount);

    }
    staticamount4() {
        console.log("4");
        this.setState({ SAmountstatus: 4, Amount: 2000 })
        console.log("4", this.state.Amount);

    }

    onclick() {
       
        this.props.navigation.navigate("PaymentDonation")
    }
    goback() {
        const { navigation }: any = this.props
        navigation.pop()
    }

    //istanbul ignore next
    async receive(from: string, message: Message) {
        
        runEngine.debugLog("on recieive==>" + JSON.stringify(message));
        
        if (getName(MessageEnum.RestAPIResponceMessage) === message.id) {
            const apiRequestCallId = message.getData(
                getName(MessageEnum.RestAPIResponceDataMessage)
            );
            let responseJson = message.getData(
                getName(MessageEnum.RestAPIResponceSuccessMessage)
            );
            const errorReponse = message.getData(
                getName(MessageEnum.RestAPIResponceErrorMessage)
            );
            if (errorReponse) {
                this.parseApiCatchErrorResponse(errorReponse);
                return
            }
            if (apiRequestCallId && responseJson) {
                if (apiRequestCallId === this.DonationGetId) {
                    this.GetDonationAPI(responseJson)
                   
                }
            }
        }
    }
    //istanbul ignore next
    GetDonationAPI(responseJson: any) {
        console.log("Deep--> ", responseJson);
        if (responseJson != null && !responseJson.error && !responseJson.errors) {
            let arr = []
            
            this.setState({
                Donation_list: responseJson
            })
            const listdata = this.state.Donation_list.donate_list.data
            for (let i = 0; i < listdata.length; i++) {
                
                let ds: any = this.state.Donation_list.donate_list.data[i].attributes.name
                
                arr.push(ds)
            }
            for (let j = 0; j < this.state.Donation_list.amount.length; j++) {
                let amt: any = this.state.Donation_list.amount[j]
                console.log(amt);
            }
            // this.setState({ Amount: amount })
            this.setState({ final_list: arr })
            console.log("FINAL DATA", this.state.final_list);
            console.log("FINAL AMOUNT", this.state.Amount);

        } else {
            console.log("error");
            setTimeout(() => {
                this.parseApiErrorResponse(responseJson);
            }, 500);
        }

    }

    GetDonationlist = () => {
        const header = {
            "Content-Type": configJSON.DonationApiContentType,
            token: this.state.token
        };

        const requestMessage = new Message(
            getName(MessageEnum.RestAPIRequestMessage)
        );

        this.DonationGetId = requestMessage.messageId;

        requestMessage.addData(
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            configJSON.DonationGetAPi
        );


        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            JSON.stringify(header)
        );

        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestMethodMessage),
            configJSON.DonationGetMethodType
        );

        runEngine.sendMessage(requestMessage.id, requestMessage);

        return true;
    }

    
}
// Customizable Area End