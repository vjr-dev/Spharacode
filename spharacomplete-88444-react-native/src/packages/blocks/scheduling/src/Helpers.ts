/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});