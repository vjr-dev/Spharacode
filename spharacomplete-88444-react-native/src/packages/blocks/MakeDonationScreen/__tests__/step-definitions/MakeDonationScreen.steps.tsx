//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'
import * as helpers from 'framework/src/Helpers'
import { Message } from "framework/src/Message"
import { runEngine } from 'framework/src/RunEngine'
import MessageEnum, { getName } from "framework/src/Messages/MessageEnum"
import React from "react";
import {Alert} from 'react-native'
import MakeDonation from "../../src/MakeDonation"
const navigation = require("react-navigation")

const screenProps = {
    navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        addListener: jest.fn().mockImplementation((event: any, callback: any) => {
            callback();
          }),
        pop:jest.fn()
    },
    state: {
      params: {
        data: { name: 'Test Name' },
        userName: 'abc'
      }
    },
    id: "MakeDonation"
  }


const feature = loadFeature('./__tests__/features/MakeDonationScreen-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'android' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'android');
    });
    test('User navigates to MakeDonationScreen', ({ given, when, then }) => {
        let MakeDonationBlockWrapper:ShallowWrapper;
        let instance:MakeDonationScreen ; 

        given('I am a User loading MakeDonationScreen', () => {
            //@ts-ignore
            MakeDonationBlockWrapper = shallow(<MakeDonation {...screenProps}/>)
        });

        when('I navigate to the MakeDonationScreen', () => {
             instance = MakeDonationBlockWrapper.instance() as MakeDonationScreen

        });

        then('MakeDonationScreen will load with out errors', () => {

            jest.spyOn(Alert, "alert").mockImplementation(() => jest.fn());
    
            instance.componentDidMount()
            instance.setState({token: "dfsffsfsdfs"})
            instance.GetDonationlist()
            instance.flatclick(2)
            instance.donatclick()
            instance.onclick()

            expect(MakeDonationBlockWrapper).toBeTruthy()
            expect(MakeDonationBlockWrapper).toMatchSnapshot()
        });
        then('fetching donation list without error', () => {
            instance.GetDonationlist()
            const DonationListAPI: Message = new Message(
                getName(MessageEnum.RestAPIResponceMessage)
              );
              DonationListAPI.addData(
                getName(MessageEnum.RestAPIResponceDataMessage),
                DonationListAPI.messageId
              );
              DonationListAPI.addData(
                getName(MessageEnum.RestAPIResponceErrorMessage),
                JSON.parse(JSON.stringify({}))
              );
              instance.getListId = DonationListAPI.messageId;
              runEngine.sendMessage("Unit Test", DonationListAPI);
        
        });

        then('I can select the button without errors', () => {
            let buttonComponent = MakeDonationBlockWrapper.findWhere((node) => node.prop('testID') === 'btnGoBack');
            
            buttonComponent.simulate('press',instance.goback())
            instance.goback()
            const doneClick = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "modaltoggleID")
             doneClick.simulate("press")
        });
        then('I can press static amount button',() => {{
            const staticClick1 = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "staticamount1ID")
            staticClick1.simulate("press",instance.staticamount1())

            const staticClick2 = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "staticamount2ID")
            staticClick2.simulate("press",instance.staticamount2())

            const staticClick3 = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "staticamount3ID")
            staticClick3.simulate("press",instance.staticamount3())

            const staticClick4 = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "staticamount4ID")
            staticClick4.simulate("press",instance.staticamount4())
          
        }});

        then('I can enter text in textbox',() => {{
            let textInputComponent1 = MakeDonationBlockWrapper.findWhere((node) => node.prop('testID') === 'customAmountID');
            textInputComponent1.simulate('changeText', instance.coustomamount("dsds"));  
               
        }});

        then('I can press donate now button',() => {{
            const doneClick = MakeDonationBlockWrapper.findWhere((node) => node.prop("testID") === "doneclickID")
            doneClick.simulate("press",instance.donatclick())
          
        }});
        
        
    })
}
)
