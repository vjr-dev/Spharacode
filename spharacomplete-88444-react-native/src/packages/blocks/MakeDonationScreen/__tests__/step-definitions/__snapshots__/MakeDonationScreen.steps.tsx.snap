// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MakeDonationScreen User navigates to MakeDonationScreen 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#f17234",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/MakeDonationScreen/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/MakeDonationScreen/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "width": 750,
          }
        }
      >
        <KeyboardAvoidingView
          behavior="width"
          enabled={true}
          keyboardVerticalOffset={1}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <StatusBar
            animated={false}
            backgroundColor="#f17234"
            barStyle="default"
            showHideTransition="fade"
          />
          <View
            style={
              Object {
                "height": 1114,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(255,138,73,1)",
                  "flexDirection": "row",
                  "height": 93.5,
                  "justifyContent": "space-between",
                  "width": 750,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "height": "90%",
                    "justifyContent": "center",
                    "width": "15%",
                  }
                }
                testID="btnGoBack"
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../packages/blocks/MakeDonationScreen/assets/image_back.png",
                    }
                  }
                  style={
                    Object {
                      "height": 15,
                      "tintColor": "#404040",
                      "width": 15,
                    }
                  }
                />
              </TouchableOpacity>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "width": "70%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#404040",
                      "fontSize": 30,
                      "fontWeight": "700",
                    }
                  }
                >
                  Make Donation
                </Text>
              </View>
              <View
                style={
                  Object {
                    "height": "100%",
                    "width": "15%",
                  }
                }
              />
            </View>
            <ScrollViewMock>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 30,
                    "justifyContent": "center",
                    "marginTop": 40,
                    "width": 730,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#cdcdcd",
                    }
                  }
                >
                  Donate to
                </Text>
              </View>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "backgroundColor": "#515151",
                    "borderRadius": 50,
                    "flexDirection": "row",
                    "height": 50,
                    "justifyContent": "space-between",
                    "width": 730,
                  }
                }
                testID="modaltoggleID"
              >
                <Text
                  style={
                    Object {
                      "color": "#a1a1a1",
                      "marginLeft": 15,
                    }
                  }
                >
                  2
                </Text>
                <Image
                  source={
                    Object {
                      "testUri": "../../../packages/blocks/MakeDonationScreen/assets/down-arrow.png",
                    }
                  }
                  style={
                    Object {
                      "height": 15,
                      "marginRight": 18,
                      "tintColor": "#a1a1a1",
                      "width": 15,
                    }
                  }
                />
              </TouchableOpacity>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "height": 30,
                    "justifyContent": "center",
                    "marginTop": 50,
                    "width": 730,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#cdcdcd",
                    }
                  }
                >
                  Amount
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "flexDirection": "row",
                    "height": 50,
                    "justifyContent": "space-between",
                    "marginTop": 20,
                    "width": 730,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#515151",
                      "borderRadius": 25,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "23%",
                    }
                  }
                  testID="staticamount1ID"
                >
                  <Text
                    style={
                      Object {
                        "color": "#a1a1a1",
                      }
                    }
                  >
                    ₹ 100
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#515151",
                      "borderRadius": 25,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "23%",
                    }
                  }
                  testID="staticamount2ID"
                >
                  <Text
                    style={
                      Object {
                        "color": "#a1a1a1",
                      }
                    }
                  >
                    ₹ 500
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#515151",
                      "borderRadius": 25,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "23%",
                    }
                  }
                  testID="staticamount3ID"
                >
                  <Text
                    style={
                      Object {
                        "color": "#a1a1a1",
                      }
                    }
                  >
                    ₹ 1000
                  </Text>
                </TouchableOpacity>
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#515151",
                      "borderRadius": 25,
                      "height": "100%",
                      "justifyContent": "center",
                      "width": "23%",
                    }
                  }
                  testID="staticamount4ID"
                >
                  <Text
                    style={
                      Object {
                        "color": "#a1a1a1",
                      }
                    }
                  >
                    ₹ 2000
                  </Text>
                </TouchableOpacity>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#515151",
                    "borderRadius": 50,
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 50,
                    "width": 730,
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  keyboardType="number-pad"
                  maxLength={10}
                  onChangeText={[Function]}
                  placeholder="Enter custom amount"
                  placeholderTextColor="#a1a1a1"
                  rejectResponderTermination={true}
                  style={
                    Object {
                      "color": "#a1a1a1",
                      "marginLeft": 20,
                    }
                  }
                  testID="customAmountID"
                  underlineColorAndroid="transparent"
                  value="0"
                />
              </View>
            </ScrollViewMock>
          </View>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "rgba(255,138,73,1)",
                "borderRadius": 50,
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 20,
                "width": 730,
              }
            }
            testID="doneclickID"
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontWeight": "700",
                }
              }
            >
              DONATE NOW
            </Text>
          </TouchableOpacity>
        </KeyboardAvoidingView>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
